cdc/processor/sinkmanager/table_sink_worker.go

```diff
@@ -109,7 +131,9 @@ func validateAndAdjustBound(changefeedID model.ChangeFeedID,
 }
 
 func (w *sinkWorker) handleTask(ctx context.Context, task *sinkTask) (finalErr error) {
-	advancer := newTableSinkAdvancer(task, w.splitTxn, w.sinkMemQuota, requestMemSize)
+	// We need to use a new batch ID for each task.
+	batchID := batchID.Add(1)
+	advancer := newTableSinkAdvancer(task, w.splitTxn, w.sinkMemQuota, requestMemSize, batchID)
```

Comment:
Why not access global `batchID` in Advancer directly?

Reply:
It breaks a lot of code and is not easy to test and maintain.

Reply:
Also, it reduces a lot of atomic operations.